src/extension/infer.rs

```diff
@@ -648,11 +648,12 @@ impl UnificationContext {
     fn search_variable_deps(&self) -> HashSet<Meta> {
         let mut seen = HashSet::new();
         let mut new_variables: HashSet<Meta> = self.variables.clone();
+        let constraints_for_solved = HashSet::new();
         while !new_variables.is_empty() {
             new_variables = new_variables
                 .into_iter()
                 .filter(|m| seen.insert(*m))
-                .flat_map(|m| self.get_constraints(&m).unwrap())
+                .flat_map(|m| self.get_constraints(&m).unwrap_or(&constraints_for_solved))
```

Comment:
AFAICS solved constraints get removed here: https://github.com/CQCL/hugr/blob/6959c8982bcce4bd8a143e12e14d54a5294c3f85/src/extension/infer.rs#L633-L635

so we have to handle them somehow. However I haven't looked into this all that hard so may have missed something @croyzor ?